src/image.ts

```diff
@@ -72,19 +73,24 @@ const sign = (salt: string, path: string): string =>
     createHash('md5').update(salt + path).digest('hex');
 
 function src(salt: string, input: string, width: number, dpr: Dpr): string {
-    const url = new URL(input);
-    const service = getSubdomain(url.hostname);
-
-    const params = new URLSearchParams({
-        width: width.toString(),
-        quality: dpr === Dpr.Two ? lowerQuality.toString() : defaultQuality.toString(),
-        fit: 'bounds',
-    });
-
-    const path = `${url.pathname}?${params.toString()}`;
-    const sig = sign(salt, path);
+    const url: Result<string, URL> = fromUnsafe(() => new URL(input), 'invalid url')
+    return pipe2(
+        toOption(url),
+        map(url => {
```

Comment:
We should probably use a different name for the parameter here to avoid shadowing the `url` in the enclosing scope.

Reply:
👍